File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        myMap = {}
        for c in s:
            myMap[c] = 1 if c not in myMap else myMap[c] + 1
        
        counter = 0
        # for c in s:
        for i, c in enumerate(s):
            if myMap[c] == 1:
                return i
        return -1

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        // create a map with char as key and instances as value
        std::unordered_map<char, int> map;
        for (int i = 0; i < s.length(); i++) {
            if (map.find(s[i]) != map.end()) {
                map[s[i]]++;
            } else {
                std::pair<char, int> p;
                p.first = s[i];
                p.second = 1;
                map.insert(p);
            }
        }
        
        // loop over chars again and check if the key has a value of 1
        for (int i = 0; i < s.length(); i++) {
            if (map[s[i]] == 1) {
                return i;
            }
        }
        return -1;
    }
};